packages/near-lake-primitives/src/types/block.ts

```diff
@@ -157,6 +166,90 @@ export class Block {
         });
     }
 
+    /**
+     * Decodes base64 encoded JSON data. Returns `undefined` if the data is not in the expected format.
+     * @param encodedValue
+     */
+    base64decode(encodedValue: any) {
+        try {
+            const buff = Buffer.from(encodedValue, "base64");
+            const str = buff.toString("utf-8").replace(/\\xa0/g, " ");
+            return JSON.parse(str);
+        } catch (error) {
+            console.error(
+              'Error parsing base64 JSON - skipping data',
+              error
+            );
+        }
+    }
+
+    /**
+     * Returns an Array of `DecodedFunctionCall` for the given `contract` and `method` if provided.
+     * If the `method` is not provided, it returns all the `DecodedFunctionCall`s for the given `contract`.
+     * Arguments to the function call are decoded from base64 to JSON.
+     * @param contract
+     * @param method
+     */
+    successfulFunctionCalls(contract: string, method?: string) : DecodedFunctionCall[] {
+        return this
+          .actions(true)
+          .filter((action) => action.receiverId === contract)
+          .flatMap((action: Action) =>
+            action.operations
+              .map((operation: Operation): FunctionCall => (operation as FunctionCallWrapper)?.FunctionCall)
+              .filter((functionCallOperation) => functionCallOperation && (!method || functionCallOperation?.methodName === method))
+              .map((functionCallOperation) => ({
+                  ...functionCallOperation,
+                  args: this.base64decode(functionCallOperation.args),
+                  receiptId: action.receiptId,
+              }))
+          );
+    };
+
+    /**
+     *  Returns data that follows the social.near contract template for key value data.
+     * @param operation The top level key to search for in each user's account space
+     * @param contract Defaults to 'social.near', pass in a different contract if needed
+     * @param method Defaults to 'set', pass in a different method if needed
+     */
+    socialOperations(operation: string, contract: string = "social.near", method: string = "set") {
+        return this.successfulFunctionCalls(contract, method)
+          .filter((functionCall) => {
+              if (
+                !functionCall ||
+                !functionCall.args ||
+                !functionCall.args.data ||
+                !Object.keys(functionCall.args.data) ||
+                !Object.keys(functionCall.args.data)[0]
+              ) {
+                  console.error(
+                    "Set operation did not have arg data in expected format"
+                  );
+                  return;
+              }
+              const accountId = Object.keys(functionCall.args.data)[0];
```

Comment:
```suggestion
              const accountId = Object.keys(functionCall.args.data ?? {})[0];
              if (!accountId) {
                  return false;
              }
```

I believe this is the same?

Same comment around logging, can we please remove? Also we should be returning `false` here and below?